Automattic/wp-calypso
# Add new 'Use a domain I own' page

#### Changes proposed in this Pull Request

This PR adds a new page that is the starting point for a new "Use a domain I own" flow to transfer or connect a domain that the customer already has elsewhere.

There will be follow-up PRs to include the rest of the new flow and hook this new page in from the domain search page.

#### Testing instructions

Use this URL: `http://calypso.localhost:3000/domains/add/use-my-domain/<site slug>?initialQuery=<domain name>`

When there is no `initialQuery` included, then the input box will be empty and it should have the focus when the page loads.

If there is an `initialQuery` included, then the input box should have this value and there should be a validation to see if it is a valid domain when the page loads.

Clicking on the "Next" button or hitting the `Enter` key should cause a validation, and go to the next page if the validation succeeds or show the error if the validation fails.

Clicking on the "X" on the right side of the input box or hitting the `Escape` key should clear the text in the input box as well as any previous error.

If there is a validation error shown, starting to type anything else should also clear the validation errors.

Screenshots:

<img width="1088" alt="Screen Shot 2021-08-05 at 11 53 18 AM" src="https://user-images.githubusercontent.com/1379730/128385959-07aa7d4a-9b55-4113-aea2-78cd274f59ec.png">

<img width="1088" alt="Screen Shot 2021-08-05 at 11 53 38 AM" src="https://user-images.githubusercontent.com/1379730/128385979-ebf84bfc-c2fb-4a43-9972-688d43ab0654.png">

<img width="1088" alt="Screen Shot 2021-08-05 at 11 53 31 AM" src="https://user-images.githubusercontent.com/1379730/128385990-6dee4ea2-68a2-45bc-a644-1528b477ba9a.png">

![screencapture-calypso-localhost-3000-domains-add-use-my-domain-a8ctest-com-2021-08-05-11_51_28](https://user-images.githubusercontent.com/1379730/128386080-fe76b865-26a1-4966-82fe-39fa6f23bbe3.png)

![screencapture-calypso-localhost-3000-domains-add-use-my-domain-a8ctest-com-2021-08-05-11_52_46](https://user-images.githubusercontent.com/1379730/128386101-1a94ec2d-8380-41ed-9114-a0e080bdf83d.png)

![screencapture-calypso-localhost-3000-domains-add-use-my-domain-a8ctest-com-2021-08-05-11_52_14](https://user-images.githubusercontent.com/1379730/128386118-c9c39b89-8dba-48f2-81bd-82873663165a.png)
